spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/procedures/GenerateSymlinkFormatManifestProcedure.java

```diff
@@ -0,0 +1,142 @@
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.Metadata;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+
+class GenerateSymlinkFormatManifestProcedure extends BaseProcedure {
+  private static final ProcedureParameter[] PARAMETERS = new ProcedureParameter[]{
+      ProcedureParameter.required("table", DataTypes.StringType),
+      ProcedureParameter.optional("symlink_root_location", DataTypes.StringType)
+  };
+
+  private static final StructType OUTPUT_TYPE = new StructType(new StructField[]{
+      new StructField("snapshot_id", DataTypes.LongType, false, Metadata.empty()),
+      new StructField("data_file_count", DataTypes.LongType, false, Metadata.empty())
+  });
+
+  private static final String PROCEDURE_CONTEXT = "generate symlink format manifest";
+  private static final String SYMLINK_DIRECTORY_DEFAULT = "_symlink_format_manifest";
+
+  private GenerateSymlinkFormatManifestProcedure(TableCatalog tableCatalog) {
+    super(tableCatalog);
+  }
+
+  public static SparkProcedures.ProcedureBuilder builder() {
+    return new Builder<GenerateSymlinkFormatManifestProcedure>() {
+      @Override
+      protected GenerateSymlinkFormatManifestProcedure doBuild() {
+        return new GenerateSymlinkFormatManifestProcedure(tableCatalog());
+      }
+    };
+  }
+
+  @Override
+  public ProcedureParameter[] parameters() {
+    return PARAMETERS;
+  }
+
+  @Override
+  public StructType outputType() {
+    return OUTPUT_TYPE;
+  }
+
+  @Override
+  public InternalRow[] call(InternalRow args) {
+    String tableIdent = args.getString(0);
+    Preconditions.checkArgument(tableIdent != null && !tableIdent.isEmpty(),
+        "Cannot handle an empty identifier for argument table");
+
+    CatalogPlugin defaultCatalog = spark().sessionState().catalogManager().currentCatalog();
+    Spark3Util.CatalogAndIdentifier catalogAndIdent = Spark3Util.catalogAndIdentifier(
+        PROCEDURE_CONTEXT, spark(), tableIdent, defaultCatalog);
+    SparkTable sparkTable = loadSparkTable(catalogAndIdent.identifier());
+    org.apache.iceberg.Table icebergTable = sparkTable.table();
+    ValidationException.check(icebergTable.currentSnapshot() != null,
+        "Cannot generate symlink manifests for an empty table");
+
+    long snapshotId = icebergTable.currentSnapshot().snapshotId();
+    String symlinkRootLocation = args.isNullAt(1) ?
+        defaultSymlinkManifestRootLocation(icebergTable.location(), snapshotId) : args.getString(1);
+
+    Types.StructType partitionType = Partitioning.partitionType(icebergTable);
+    Dataset<Row> entries = SparkTableUtil.loadCatalogMetadataTable(spark(), icebergTable, MetadataTableType.ENTRIES)
+        .filter("status < 2 AND data_file.content = 0");
+
+    long rowCount = entries.count();
+    if (partitionType.fields().isEmpty()) {
+      entries.select("data_file.file_path")
+          .write()
+          .format("parquet")
```

Comment:
[question] As per this : https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/io/SymlinkTextInputFormat.java#L47-L52

**_Symlink file is a text file which contains a list of filename / dirname_**

does it works when Symlink file is in parquet format as well ? Am i missing something here